rbac/management/group/relation_api_dual_write_group_handler.py

```diff
@@ -123,39 +129,55 @@ def _replicate(self):
         except Exception as e:
             raise DualWriteException(e)
 
-    def replicate_added_role(self, role: Role):
-        """Replicate added role."""
+    def generate_relations_to_add_roles(
+        self, roles: Iterable[Role], remove_default_access_from: Optional[TenantMapping] = None
+    ):
+        """Generate relations to add roles."""
         if not self.replication_enabled():
             return
 
         def add_group_to_binding(mapping: BindingMapping):
             self.group_relations_to_add.append(mapping.add_group_to_bindings(str(self.group.uuid)))
 
-        def create_default_mapping():
-            assert role.system is True, "Expected system role. Mappings for custom roles must already be created."
+        def create_default_mapping(system_role):
+            assert system_role.system is True, "Expected system role. Mappings for custom roles must already be created."
             binding = V2rolebinding(
                 str(uuid4()),
                 # Assumes same role UUID for V2 system role equivalent.
-                V2role.for_system_role(str(role.uuid)),
+                V2role.for_system_role(str(system_role.uuid)),
                 V2boundresource(("rbac", "workspace"), str(self.default_workspace.id)),
                 groups=frozenset([str(self.group.uuid)]),
             )
-            mapping = BindingMapping.for_role_binding(binding, role)
+            mapping = BindingMapping.for_role_binding(binding, system_role)
             self.group_relations_to_add.extend(mapping.as_tuples())
             return mapping
```

Comment:
I think we can move this just to be a top-level function now if you want (make it private in that case e.g. with _ prefix)